Popup/tests/Popup-specs.js

```diff
@@ -1,180 +1,181 @@
 	describe('with position center', function () {
 		test('should have position=center when no position is specified', () => {
-			const popup = mount(
-				<PopupBase />
+			render(
+				<FloatingLayerController>
+					<Popup open />
+				</FloatingLayerController>
 			);
 
 			const expected = 'center';
-			const actual = popup.prop('position');
+			const popup = screen.getByRole('alert');
 
-			expect(actual).toBe(expected);
+			expect(popup).toHaveClass(expected);
 		});
 	});
 
 	describe('with position top', function () {
 		test('should have top class', () => {
-			const popup = shallow(
-				<PopupBase open position="top">
-					<div>popup</div>
-				</PopupBase>
+			render(
+				<FloatingLayerController>
+					<Popup open position="top"><div>popup</div></Popup>
+				</FloatingLayerController>
 			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).toContain('top');
+			const expected = 'top';
+			const popup = screen.getByRole('alert');
+
+			expect(popup).toHaveClass(expected);
 		});
 	});
 
 	describe('with position changes dynamically', function () {
 		test('should not have top class when position change from top to any other position', () => {
-			const firstPosition = 'top';
-			const popup = shallow(
-				<PopupBase open position={firstPosition}>
-					<div>popup</div>
-				</PopupBase>
+			const {rerender} = render(
+				<FloatingLayerController>
+					<Popup open position="top"><div>popup</div></Popup>
+				</FloatingLayerController>
 			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).toContain(firstPosition);
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('center');
-			expect(popup.prop('className').split(' ')).not.toContain('center');
+			const firstExpected = 'top';
+			const initialPopup = screen.getByRole('alert');
 
-			popup.setProps({position: 'center'});
-
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain(firstPosition);
-			expect(popup.prop('className').split(' ')).not.toContain(firstPosition);
-		});
-	});
-
-	test('should have `center` class when position prop is set to `center` (default)', () => {
-		const popup = shallow(
-			<PopupBase position="center" />
-		);
+			expect(initialPopup).toHaveClass(firstExpected);
 
-		const expected = 'center';
-		const actual = popup.find(`.${css.popup}`).prop('className');
-
-		expect(actual).toContain(expected);
-	});
-
-	test('should have `top` class when position prop is set to `top`', () => {
-		const popup = shallow(
-			<PopupBase position="top" />
-		);
+			rerender(
+				<FloatingLayerController>
+					<Popup open position="center"><div>popup</div></Popup>
+				</FloatingLayerController>
+			);
 
-		const expected = 'top';
-		const actual = popup.find(`.${css.popup}`).prop('className');
+			const secondExpected = 'center';
+			const popup = screen.getByRole('alert');
 
-		expect(actual).toContain(expected);
+			expect(popup).toHaveClass(secondExpected);
```

Comment:
The result does not really reflect the test's description.

Reply:
The test is split in two. First we render the popup with position top and then we rerender the popup with position center. I left a comment in code for better understanding.